bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java

```diff
@@ -232,31 +240,44 @@ private void updateZoneAlarm(WhatAlarm w) {
         switch (w) {
             case ZONE_ALARM_INTRUSION:
                 updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_INTRUSION));
+                updateZoneAlarmTimeStamp();
                 break;
             case ZONE_ALARM_TAMPERING:
                 updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_TAMPERING));
+                updateZoneAlarmTimeStamp();
                 break;
             case ZONE_ALARM_ANTI_PANIC:
                 updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_ANTI_PANIC));
+                updateZoneAlarmTimeStamp();
                 break;
             case ZONE_ALARM_SILENT:
                 updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_SILENT));
+                updateZoneAlarmTimeStamp();
                 break;
             case ZONE_ALARM_TECHNICAL:
                 updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_TECHNICAL));
+                updateZoneAlarmTimeStamp();
                 break;
             case ZONE_ALARM_TECHNICAL_RESET:
                 updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_TECHNICAL_RESET));
+                updateZoneAlarmTimeStamp();
                 break;
             default:
                 updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_NONE));
                 logger.warn("Alarm.updateZoneAlarm() Ignoring unsupported WHAT {} for  zone {}", w, this.deviceWhere);
         }
     }
 
+    private void updateZoneAlarmTimeStamp() {
+        ZonedDateTime now = ZonedDateTime.now();
+        ChannelUID cuid = new ChannelUID(getThing().getUID(), CHANNEL_ALARM_ZONE_TIMESTAMP);
```

Comment:
this line is useless. Why don't you do the same like in initialize():
`updateState(CHANNEL_ALARM_ZONE_TIMESTAMP, ....`

Reply:
at this point could I call updateTimeStmp()?